### DISCUSSION TITLE
Layered lifebars

### DISCUSSION THREAD

**Discussion by DolphinDan400** [ikemen-engine/Ikemen-GO#897] [Feature Request]
I'd be cool to see Lifebar support for multiple front elements layering on top of each other (usually 2 or 3 layers), each representing a fraction of the character's HP similar to the levelbars feature of the Powerbar, if possible with layers being divided automatically depending per character max life stat.

Some examples I can think of are Real Bout Fatal Fury, Advanced Variable Geo 2 and SVC Chaos.
![layerbars](https://user-images.githubusercontent.com/94821269/158901260-17ba2f83-c685-4bda-9e7b-dfedfc9a1616.png)

**Comment by ghost** [comment]
This should be a fairly simple implementation, imo.
I may take a look at the go code and find a way to implement this.

**Comment by MUGEN-Slayer** [comment]
I remember Gacel mentioning this at some point.
I even made a design of how it might work in fight.def, but I never shared it for lack of opportunity. So I'll leave it here just in case:

How thing works right now:

;Player 1
p1.pos    = 320,0
p1.bg0.anim = 0
p1.bg1.spr = 1,0
p1.bg2.spr = 2,0
p1.mid.spr = 3,0
p1.front.spr = 4,0
p1.range.x  = -6,-433

We could have something like this:

p1.tank0.pos    = 320,0
p1.tank0.bg0.anim = 0
p1.tank0.bg1.spr = 1,0
p1.tank0.bg2.spr = 2,0
p1.tank0.mid.spr = 3,0
p1.tank0.front.spr = 4,0
p1.tank0.range.x  = -6,-433
p1.tank0.enabled  = 50,0

p1.tank1.pos    = 320,0
p1.tank1.bg0.anim = 10
p1.tank1.bg1.spr = 11,0
p1.tank1.bg2.spr = 12,0
p1.tank1.mid.spr = 13,0
p1.tank1.front.spr = 14,0
p1.tank1.range.x  = -6,-433
p1.tank1.enabled  = 100,51


Tank1 would overlap tank0 to keep the consistency on how things work so far.

And for the sake of compatibility with old bars: "p1.tank0.pos" = "p1.pos". 


Edit: I forgot a parameter to indicate when to switch between tanks so I added it. For example:

p1.tank0.enabled  = 50,0
p1.tank1.enabled  = 100,51

With "tank1" overlapping "tank0", tank 1 would be active as long as the player's health is between 100% and 51%, while tank0 would work from 50% to 0%.

**Comment by Kuroteba** [comment]
Is it possible to make the timing of tank switching a real number instead of a percentage of life, as in Kirby Fighters 2?
